### ISSUE TITLE
Add `.editorconfig`

### ISSUE THREAD

**Issue by akaihola** [akaihola/darglint2#14]
> I also propose having an [`.editorconfig`](https://editorconfig.org/) and maybe a `.gitattributes` file configuring the diff algo for different file types.

_Originally posted by @real-yfprojects in [a comment to issue #1](https://github.com/akaihola/darglint2/issues/1#issuecomment-1437534712_)

**Comment by akaihola** [comment]
@real-yfprojects thanks for the idea! Could you elaborate what this addition brings for the `darglint2` linter tool in concrete terms?

**Comment by real-yfprojects** [comment]
Basically it tells you favourite IDE how to format your files. This ensures consistent formatting and reduces related clutter in commits. Furthermore it reduces configuration effort for new contributors and the amount of work needed by maintainers fix PR formatting. EditorConfig supports [these basic properties](https://editorconfig.org/#supported-properties).

**Comment by akaihola** [comment]
I took a look at [the basic properties](https://editorconfig.org/#supported-properties) supported by EditorConfig. Here are some observations:

- **indent_style**: Black enforces spaces, so Darker needs to fail if this is set to `"tab"`.
- **indent_size**: Black enforces four spaces, so Darker needs to fail if this is not set to `4`.
- **tab_width**: tabs _might_ appear in comments, strings or even at the beginning of the line in multi-line strings, and Black won't (probably?) touch those, so while this would affect the IDE, Darker really doesn't need to care
- **end_of_line**: Black and Darker make the best effort to output the same newline characters it sees in its input, and I prefer nothing else affecting that behavior. If the IDE changes the newline style, Darker will adapt.
- **charset**: Darker detects the encoding of input files and uses the same encoding in its output. Again, I prefer nothing else affecting this behavior. If the IDE changes the encoding, Darker will adapt.
- **trim_trailing_whitespace**: Black strips trailing whitespace except in multi-line strings. Let's assume an IDE follows this EditorConfig literally. In the  '_`"true"` to remove any whitespace characters preceding newline characters_' case, Black won't have any trailing whitespace to remove, and all is fine. But in the '_`"false"` to ensure it doesn't_' case, we can't stop Black from removing trailing whitespace, and Darker must fail (at least if trailing whitespace exists).
- **insert_final_newline**: Black will always insert a missing final newline, so Darker needs to fail if this is set to `"false"` (at least if a missing final newline is encountered)

To summarize:
- `.editorconfig` can't affect Darker's behavior
- Darker should merely validate that `.editorconfig` matches Black behavior for `.py` files, and fail if any of the following options are encountered:
```ini
indent_style = tab
indent_size = <N>  # if <N> ≠ 4
trim_trailing_whitespace = false  # or fail at least if trailing whitespace is found
insert_final_newline = false  # or fail at least if a missing final newline is encountered
```

**Comment by akaihola** [comment]
@real-yfprojects, sorry, I was working on Darker and apparently was too tired (or enthusiastic) to notice that this was actually about `darglint2`... 

So an `.editorconfig` with settings which correspond to the style of the `darglint2` code base?

**Comment by real-yfprojects** [comment]
Yes basically, see ab8dbd3ec66820749d651406f6c372fb6b838993.
